src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -196,6 +196,19 @@ cmd_state_to_string(ert_cmd_state state)
     : itr->second;
 }
 
+std::string
+fatal_err_type_to_string(uint32_t fatal_error_type)
```

Comment:
This function is not obvious given that it's argument type is a uint32_t.   What is the argument?  Who would want to call this function.    I think this is very specific to error_impl, so make it a private member function of that class.